bee-network/src/peer/list.rs

```diff
@@ -0,0 +1,456 @@
+                if predicate(info, state) { count + 1 } else { count }
+            },
+        )
+    }
+
+    pub fn remove_if<P>(&mut self, peer_id: &PeerId, predicate: P) -> bool
+    where
+        P: Fn(&PeerInfo, &PeerState) -> bool,
+    {
+        let can_remove = if let Some((info, state)) = self.peers.get(peer_id) {
+            predicate(info, state)
+        } else {
+            return false;
+        };
+
+        if can_remove {
+            self.peers.remove(peer_id).is_some()
+        } else {
+            false
+        }
+    }
+
+    #[cfg(test)]
+    pub fn clear(&mut self) {
+        self.peers.clear();
+        self.banned_peers.clear();
+        self.banned_addrs.clear();
+    }
+
+    pub fn ban_peer(&mut self, peer_id: PeerId) -> Result<(), Error> {
+        // TODO: use storage to persist banned peers
+        if self.banned_peers.insert(peer_id) {
+            Ok(())
+        } else {
+            Err(Error::PeerIsBanned(peer_id))
+        }
+    }
+
+    pub fn ban_address(&mut self, address: Multiaddr) -> Result<(), Error> {
+        // TODO: use storage to persist banned addrs
+        if self.banned_addrs.insert(address.clone()) {
+            Ok(())
+        } else {
+            Err(Error::AddressIsBanned(address))
+        }
+    }
+
+    pub fn unban_peer(&mut self, peer_id: &PeerId) -> Result<(), Error> {
+        if self.banned_peers.remove(peer_id) {
+            Ok(())
+        } else {
+            Err(Error::PeerIsUnbanned(*peer_id))
+        }
+    }
+
+    pub fn unban_address(&mut self, address: &Multiaddr) -> Result<(), Error> {
+        if self.banned_addrs.remove(address) {
+            Ok(())
+        } else {
+            Err(Error::AddressIsUnbanned(address.clone()))
+        }
+    }
+
+    pub fn is_peer_banned(&self, peer_id: &PeerId) -> bool {
+        self.banned_peers.contains(peer_id)
+    }
+
+    pub fn is_addr_banned(&self, address: &Multiaddr) -> bool {
+        self.banned_addrs.contains(address)
+    }
+
+    pub fn accepts_incoming_peer(&self, peer_id: &PeerId, peer_info: &PeerInfo) -> Result<(), Error> {
+        if peer_id == &self.local_id {
+            return Err(Error::PeerIsLocal(*peer_id));
+        }
+
+        if self.local_addrs.contains(&peer_info.address) {
+            return Err(Error::AddressIsLocal(peer_info.address.clone()));
+        }
+
+        if self.banned_peers.contains(peer_id) {
+            return Err(Error::PeerIsBanned(*peer_id));
+        }
+
+        if self.banned_addrs.contains(&peer_info.address) {
+            return Err(Error::AddressIsBanned(peer_info.address.clone()));
+        }
+
+        if self.is(peer_id, |_, state| state.is_connected()).unwrap_or(false) {
+            return Err(Error::PeerIsConnected(*peer_id));
+        }
+
+        if peer_info.relation.is_unknown() && self.count_if(|info, _| info.relation.is_unknown()) >= max_unknown_peers()
+        {
+            return Err(Error::ExceedsUnknownPeerLimit(max_unknown_peers()));
```

Comment:
This `return` is the penultimate branch in this function. May I suggest refactoring this to be an if-else chain to make it more obvious that it's performing a filtering operation?

Reply:
done